R/run-nmtran.R

```diff
@@ -0,0 +1,330 @@
+  # Check if nmtran_presort exists
+  run_presort <- !is.null(nmtran_presort_exe)
+
+  # Preprocess with nmtran_presort
+  if(run_presort){
+    presort.p <- processx::run(
+      command = nmtran_presort_exe, wd = run_dir, args = character(),
+      stdout = "|", stdin = file.path(run_dir, mod_path),
+      stderr_to_stdout = TRUE, error_on_status = FALSE
+    )
+
+    presort_status_val <- presort.p$status
+    if(is.na(presort_status_val)){
+      rlang::abort("nmtran_presort terminated unexpectedly")
+    }else if(presort_status_val != 0){
+      return(
+        list(
+          nmtran_model = file.path(run_dir, mod_path),
+          run_dir = run_dir,
+          status = "nmtran_presort failed. See errors.",
+          status_val = presort_status_val,
+          output_lines = presort.p$stdout
+        )
+      )
+    }
+
+    # Write the output to tempzzzz1 control stream
+    writeLines(presort.p$stdout, con = file.path(run_dir, "tempzzzz1.ctl"))
+  }
+
+  # Run NM-TRAN
+  stdin_file <- ifelse(run_presort, "tempzzzz1.ctl", mod_path)
+  nmtran.p <- processx::run(
+    command = nmtran_exe, wd = run_dir, args = cmd_args,
+    stdout = "|", stdin = file.path(run_dir, stdin_file),
+    stderr_to_stdout = TRUE, error_on_status = FALSE
+  )
+
+  # Assign status
+  status_val <- nmtran.p$status
+  if(is.na(status_val)){
+    rlang::abort("NM-TRAN terminated unexpectedly")
+  }else if(status_val == 0){
+    status <- "NM-TRAN successful"
+  }else{
+    status <- "NM-TRAN failed. See errors."
+  }
+
+  # Tabulate NM-TRAN results
+  nmtran_results <- list(
+    nmtran_model = stdin_file,
+    run_dir = run_dir,
+    status = status,
+    status_val = status_val,
+    output_lines = nmtran.p$stdout
+  )
+
+  return(nmtran_results)
+}
+
+
+#' Helper for finding and reading in a `bbi.yaml` file given a `bbi_nonmem_model`
+#' object or explicit `.config_path`.
+#' @inheritParams run_nmtran
+#' @noRd
+get_bbi_yaml_path <- function(.mod = NULL, .config_path = NULL){
+  if(!is.null(.mod)){
+    check_model_object(.mod, "bbi_nonmem_model")
+    model_dir <- get_model_working_directory(.mod)
+  }
+  bbi_yaml_path <- .config_path %||% file.path(model_dir, "bbi.yaml")
+
+  if(!file_exists(bbi_yaml_path)){
+    if(is.null(.config_path)){
+      msg <- c(
+        "No bbi configuration was found in the execution directory.",
+        "i" = "Please run `bbi_init()` with the appropriate directory to continue."
+      )
+    }else{
+      msg <- glue("No bbi configuration was found at {.config_path}")
+    }
+    rlang::abort(msg)
+  }
+
+  return(bbi_yaml_path)
+}
+
+
+#' Parse `.bbi_args` and return the three expected `nmfe_options` for `NM-TRAN`
+#'  in the correct format
+#' @inheritParams run_nmtran
+#' @param nmfe_options named list of nmfe options defined in a `bbi.yaml` file.
+#' @noRd
+parse_nmtran_args <- function(
+    .mod,
+    .bbi_args = NULL,
+    nmfe_options = NULL
+){
+
+  nmfe_args_def <- list(prdefault = FALSE, tprdefault = FALSE, maxlim = 2)
```

Comment:
It would be good to explain these defaults in code comment or the noRd docstring above. Basically, explain that these are constructed by studying what `nmfe*` passes to NM-TRAN.

---

bbi v3.2.0 changed its maxlim default to 2: [b30413e](https://github.com/metrumresearchgroup/bbi/commit/b30413e847c82f5ed6365bfcdafdbf2c7ad6f234) (nonmem: pass -maxlim=2 to nmfe by default, 2022-08-01, gh-279). I don't think we should add a compatibility kludge for this, and it doesn't necessarily deserve a code comment. But I'll mention it here because I think it's worth at least mentioning somewhere in this PR.